shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/metadata/DataSourceMetaDataTest.java

```diff
@@ -0,0 +1,184 @@
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertIsInSameDatabaseInstanceWithOracle() {
+        OracleDataSourceMetaData actual1 = new OracleDataSourceMetaData("jdbc:oracle:thin:@//127.0.0.1:9999/ds_0", "test");
+        OracleDataSourceMetaData actual2 = new OracleDataSourceMetaData("jdbc:oracle:thin:@//127.0.0.1:9999/ds_1", "test");
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertIsInSameDatabaseInstanceWithMariaDB() {
+        MariaDBDataSourceMetaData actual1 = new MariaDBDataSourceMetaData("jdbc:mariadb://127.0.0.1:9999/ds_0?serverTimezone=UTC&useSSL=false");
+        MariaDBDataSourceMetaData actual2 = new MariaDBDataSourceMetaData("jdbc:mariadb://127.0.0.1:9999/ds_1?serverTimezone=UTC&useSSL=false");
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertIsInSameDatabaseInstanceWithPostgreSQL() {
+        PostgreSQLDataSourceMetaData actual1 = new PostgreSQLDataSourceMetaData("jdbc:postgresql://127.0.0.1/ds_0");
+        PostgreSQLDataSourceMetaData actual2 = new PostgreSQLDataSourceMetaData("jdbc:postgresql://127.0.0.1/ds_1");
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertIsInSameDatabaseInstanceWithSQL92() {
+        SQL92DataSourceMetaData actual1 = new SQL92DataSourceMetaData("jdbc:sql92_db:ds_0");
+        SQL92DataSourceMetaData actual2 = new SQL92DataSourceMetaData("jdbc:sql92_db:ds_1");
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertIsInSameDatabaseInstanceWithSQLServer() {
+        SQLServerDataSourceMetaData actual1 = new SQLServerDataSourceMetaData("jdbc:microsoft:sqlserver://127.0.0.1:9999;DatabaseName=ds_0");
+        SQLServerDataSourceMetaData actual2 = new SQLServerDataSourceMetaData("jdbc:microsoft:sqlserver://127.0.0.1:9999;DatabaseName=ds_1");
+        assertTrue(actual1.isInSameDatabaseInstance(actual2));
+    }
+
+    @Test
+    public void assertFalseIsInSameDatabaseInstanceWithMysqlAndH2() {
+        H2DataSourceMetaData actualH2 = new H2DataSourceMetaData("jdbc:h2:mem:ds_0;DB_CLOSE_DELAY=-1;DATABASE_TO_UPPER=false;MODE=MySQL");
+        MySQLDataSourceMetaData actualMysql = new MySQLDataSourceMetaData("jdbc:mysql://127.0.0.1:9999/ds_0?serverTimezone=UTC&useSSL=false");
+        assertFalse(actualMysql.isInSameDatabaseInstance(actualH2));
+    }
+
+    @Test
+    public void assertFalseIsInSameDatabaseInstanceWithMysqlAndOracle() {
+        OracleDataSourceMetaData actualOracle = new OracleDataSourceMetaData("jdbc:oracle:thin:@//127.0.0.1:9988/ds_0", "test");
+        MySQLDataSourceMetaData actualMysql = new MySQLDataSourceMetaData("jdbc:mysql://127.0.0.1:9999/ds_0?serverTimezone=UTC&useSSL=false");
+        assertFalse(actualMysql.isInSameDatabaseInstance(actualOracle));
+    }
```

Comment:
We could simplify following similar test cases, just cover more code branches to improve test coverage.